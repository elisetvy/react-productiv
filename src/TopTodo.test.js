import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

describe("Top todo component", function (){
  const initialTodos=[
    {
      id: 1,
      title: "Code!",
      description: "Write some code",
      priority: 2,
    },
    {
      id: 2,
      title: "Make dinner",
      description: "Cook something healthy",
      priority: 1,
    },
    {
      id: 3,
      title: "Go to bed",
      description: "In bed by 11:15",
      priority: 3,
    },
  ]

  test("Should render without crashing", function () {
    render(<TopTodo todos={initialTodos} />);
  });

  test("contains expected title", function () {
    const result = render(<TopTodo todos={initialTodos} />);
    expect(result.queryByText("Cook something healthy")).toBeInTheDocument();
  });

  test("passes snapshot", function (){
    const { container } = render(<TopTodo todos={initialTodos} />);
    expect(container).toMatchSnapshot();
  });
});