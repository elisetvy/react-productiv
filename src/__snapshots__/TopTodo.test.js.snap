// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component passes snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        test
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        moretest
      </small>
    </div>
  </div>
</div>
`;

exports[`Top todo component passes snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="2"
  >
    <div>
      <b>
        Make dinner
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Cook something healthy
      </small>
    </div>
  </div>
</div>
`;
